client/actions/graphql/interestGQLActions.js

```diff
@@ -1,24 +1,49 @@
 import INTERESTS_LIST_GQL from '../../Graphql/Queries/InterestsListGQL';
-
-import { INTEREST, INTERESTS } from '../constants';
-
-import { handleError } from '../../utils/errorHandler';
+import CREATE_INTEREST_GQL from '../../Graphql/Mutations/CreateInterestGQL';
+import REMOVE_INTEREST_GQL from '../../Graphql/Mutations/RemoveInterestGQL';
+import { INTEREST, INTERESTS, CREATE_INTERESTS, REMOVE_INTERESTS } from '../constants';
+import { handleError, handleInformation } from '../../utils/errorHandler';
 import Client from '../../client';
 
 
 export const getInterestsList = (before = '', after = '', first = 1, last = 1) => dispatch => Client.query(
-  INTERESTS_LIST_GQL(before, after, first, last)
-).then(data => dispatch({
-  type: INTERESTS,
-  payload: data.data,
-  error: false
-}))
-.catch(error => handleError(error, dispatch));
+  INTERESTS_LIST_GQL()
```

Comment:
Please revise this GraphQL query. It is querying `InterestList` instead of `JoinedCategories`. Since we are trying to fetch the user's current interests, the `joinedCategories` query is more appropriate. This for example is making the initial behavior of the Interests page quite weird as it activates interests a user doesn't have.
`
For context, the InterestList query returns all the interests in the database, which is not what we want
`

Reply:
Done